### PR TITLE
NIFI-1218 upgraded Kafka to 0.9.0.0 client API

### PR DESCRIPTION
**Pull request by olegz**
Tested and validated that it is still compatible with 0.8.* Kafka brokers

### CODE REVIEW

File: nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java [hunk 1]
```diff
@@ -474,6 +475,18 @@ public void setStopFailingAfter(final Integer stopFailingAfter) {
         @Override
         public void close() {
         }
+
+        @Override
+        public void close(long arg0, TimeUnit arg1) {
+            // TODO Auto-generated method stub
```
Comment (JPercivall) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java]: This should either be implemented or change the comment to say why it wasn't.

File: nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java [hunk 2]
```diff
@@ -474,6 +475,18 @@ public void setStopFailingAfter(final Integer stopFailingAfter) {
...
+
+        }
+
+        @Override
+        public void flush() {
+            // TODO Auto-generated method stub
```
Comment (JPercivall) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java]: This should either be implemented or change the comment to say why it wasn't.

Comment (olegz) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java]: As you can see, this is a MockProducer defined in test. Those two methods are only there to comply with the changes in Kafka interface and are not part of the test assertions.

Comment (olegz) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java]: Having said that I'll add a comments for clarity

Comment (JPercivall) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/test/java/org/apache/nifi/processors/kafka/TestPutKafka.java]: Right, and that's a good reason why but it should be documented there.

**Comment by olegz** [comment]
Addressed

**Comment by olegz** [comment]
Merged

**Pull request closed without merging**